client/src/util/renderThrottle.ts

```diff
@@ -1,17 +1,16 @@
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-export default function renderThrottle(callback: any) {
+export default function renderThrottle(
```

Comment:
I don't think this is right, as it doesn't preserve the type of `this`.  I believe you want something like:

```
export default function renderThrottle<T>(
  callback: (this: T) => void
): (this: T) => void {
  /*
  This wraps a call to requestAnimationFrame(), enforcing a single
  render callback at any given time (ie, you can call this any number
  of times, and it will coalesce multiple inter-frame calls into a
  single render).
  */
  let rafCurrentlyInProgress: number | null = null;
  return function f(this: T) {
    if (rafCurrentlyInProgress) return;
    rafCurrentlyInProgress = window.requestAnimationFrame(() => {
      callback.call<T, never, void>(this);
      rafCurrentlyInProgress = null;
    });
  };
}
```

You can test this with something like the following:
```
function bar(this: number) {
  console.log(this);
}
const foo = renderThrottle(bar);
foo.call("hi");
foo.call(1);
```

(the first `foo.call` should get a type error).

Two changes here: 
1. use a lambda which preserves `this` (so you don't need to alias it)
2. explicitly declare `this` type.

Reply:
Ah actually should we just use `lodash.throttle` since by default it uses `rAF` too? 

Also not sure if it matters, but their implementation will invoke the callback with its arguments in addition to passing `this`

https://github.com/lodash/lodash/pull/3567/files#diff-277bea254c6f5a290a109a6cb6c796e89ef069c7964c3ac8095808d779630682R98

Reply:
My view: try to minimize code changes unless you are committing to writing tests, absorbing the bugs, etc.  We don't need the args in our use cases (we pass state by `this`).

Also, historically we are on a long-term mission to remove lodash because of bloat.  Before introducing new dependencies, might want to discuss that.

KIS!

Reply:
pretty funny that lodash gave in and just typed args as any lol 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/d3kit/index.d.ts#L156

```ts
function throttle(fn: (...args: any[]) => void, delay: number): (...args: any[]) => void;
```

Reply:
In general, we just haven't found lodash to be all that useful in most situations (eg, look at how they typed memoize).  In many cases, our code doesn't want the lodash generality, OR we prefer native (eg, don't use lodash map/filter/....)

I'm not making a strong argument, other than "why change the implementation when you are just adding types"?